storage/aptosdb/src/pruner/transaction_store/test.rs

```diff
@@ -27,17 +32,190 @@ proptest! {
+            aptos_db,
+            first_available_version - 1,
+            ledger_version,
+            transactions,
+        );
+    }
+
+    // Verify no panic for a version which is present in the DB
+    if first_available_version <= ledger_version {
+        verify_no_panic_for_apis_for_present_version(
+            aptos_db,
+            first_available_version,
+            ledger_version,
+            transactions,
+        );
+    }
+}
+
+fn verify_no_panic_for_apis_for_present_version(
+    db: &AptosDB,
+    version: Version,
+    ledger_version: Version,
+    transactions: &[TransactionToCommit],
+) {
+    assert!(db
+        .get_transactions(version, 1, ledger_version, true)
+        .is_ok());
+
+    assert!(db
+        .get_transaction_by_hash(
+            transactions[version as usize].transaction().hash(),
+            ledger_version,
+            true,
+        )
+        .unwrap()
+        .is_some());
+
+    for event in transactions[version as usize].events() {
+        assert!(!db
+            .get_events(event.key(), version, Order::Ascending, 1)
+            .unwrap()
+            .is_empty());
+        assert!(!db
+            .get_events_with_proofs(event.key(), version, Order::Ascending, 1, None)
+            .unwrap()
+            .is_empty());
+        assert!(db
+            .get_event_by_version_with_proof(event.key(), version, version)
+            .unwrap()
+            .lower_bound_incl
+            .is_some());
+    }
+    assert!(db
+        .get_transaction_by_version(version, ledger_version, true)
+        .is_ok());
+    assert!(db.get_first_txn_version().is_ok());
+    assert!(db.get_first_write_set_version().is_ok());
+    assert!(db
+        .get_transaction_outputs(version, 10, ledger_version)
+        .is_ok());
+    assert!(db.get_block_timestamp(version).is_ok());
+    assert!(db.get_latest_version().is_ok());
+    assert!(db.get_latest_commit_metadata().is_ok());
+}
+
+fn verify_no_panic_for_apis_for_not_present_version(
+    db: &AptosDB,
+    version: Version,
+    ledger_version: Version,
+    transactions: &[TransactionToCommit],
+) {
+    assert!(db
+        .get_transactions(version, 10, ledger_version, true)
+        .is_err());
+
+    assert!(db
+        .get_transaction_by_hash(
+            transactions[version as usize].transaction().hash(),
+            ledger_version,
+            true,
+        )
+        .unwrap()
+        .is_none());
+
+    for event in transactions[version as usize].events() {
+        assert!(db
+            .get_events(event.key(), version, Order::Ascending, 1)
+            .unwrap()
+            .is_empty());
+        assert!(db
+            .get_events_with_proofs(event.key(), version, Order::Ascending, 1, None)
+            .unwrap()
+            .is_empty());
+        assert!(db
+            .get_event_by_version_with_proof(event.key(), version, version)
+            .unwrap()
+            .lower_bound_incl
+            .is_none());
```

Comment:
I think it makes more sense if these APIs return errors when the version is pruned, would you agree? 
Could you check if it's easy to make it so?

Reply:
The API behavior seems expected to me. `EventByVersionWithProof` has Option<EventWithProof>, so it is reasonable to return empty result without throwing errors.

Reply:
This is also inline with behavior of other events APIs like `get_events_with_proofs` and `get_events` where absent of  the event returns empty result without throwing an error.

Reply:
I mean, there is a difference between "the event doesn't exist" and "the event existed but pruned" or "we don't know if the event existed".